src/configuration/ApiConfigurationParser.ts

```diff
@@ -0,0 +1,18 @@
+import { parseString } from "xml2js";
+import { DefaultApiConfiguration, ApiConfiguration } from "configuration/apiConfiguration";
+
+export interface ApiConfigurationParser {
+    parse(xmlConfig: string): Promise<ApiConfiguration>;
```

Comment:
Do we have much to gain from the use of Promises here?
As far as I understand we can't do much until the configuration is loaded, so why not doing it without a Promise? Added bonus: simple synchronous code instead of async one.